### ISSUE TITLE
Azure OpenAI Compatibility

### ISSUE THREAD

**Issue by Tansif** [neo4j/neo4j-graphrag-python#190]
Does it support Azure OpenAI LLM and Embeddings models?

**Comment by stellasia** [comment]
Hi @Tansif ,

Yes to both:

```python
from neo4j_graphrag.embeddings import AzureOpenAIEmbeddings

embeder = AzureOpenAIEmbeddings(
    model="text-embedding-ada-002",
    azure_endpoint="https://my-endpoint.openai.azure.com/",
    api_key="<my key>",
    api_version="<update version>",
)
```

and 

```python
from neo4j_graphrag.llm.openai_llm import AzureOpenAILLM
```

**Comment by brandon-amg** [comment]
Hello @stellasia!

I receive the following error when trying to instantiate the AzureOpenAIEmbeddings object.

Can you confirm on your end?

Thanks!

## Error
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "neo4j_graphrag\embeddings\openai.py", line 69, in __init__   
    super().__init__(model, **kwargs)
  File "neo4j_graphrag\embeddings\openai.py", line 51, in __init__   
    self.openai_client = openai.OpenAI(**kwargs)
                         ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: OpenAI.__init__() got an unexpected keyword argument 'azure_endpoint'
```
## Work Around
I have been able to work around this error by making the following changes to the **neo4j_graphrag\embeddings\openai.py** file.

**Changes Made**
1. AzureOpenAIEmbeddings inherits from the BaseOpenAIEmbeddings class vs. OpenAIEmbeddings class
2. Implement the embed_query method for the AzureOpenAIEmbeddings class required by BaseOpenAIEmbeddings

### Modified Working Code
```python
class AzureOpenAIEmbeddings(BaseOpenAIEmbeddings):
    def __init__(self, model: str = "text-embedding-ada-002", **kwargs: Any) -> None:
        super().__init__(model, **kwargs)
        self.openai_client = openai.AzureOpenAI(**kwargs)
        
    def embed_query(self, text: str, **kwargs: Any) -> list[float]:
        """
        Generate embeddings for a given query using a OpenAI text embedding model.

        Args:
            text (str): The text to generate an embedding for.
            **kwargs (Any): Additional arguments to pass to the OpenAI embedding generation function.
        """
        response = self.openai_client.embeddings.create(
            input=text, model=self.model, **kwargs
        )
        return response.data[0].embedding
```

**Comment by stellasia** [comment]
You're right, there is an issue here, we are trying to instantiate the `OpenAI` client because of the `super` call. Would you mind opening a PR to fix it? The only comment I'd have on your changes, is that the `embed_query` can probably be part of the `BaseOpenAIEmbeddings` (no need to repeat it).

**Comment by willtai** [comment]
Closing this issue as this has been addressed in neo4j-graphrag version 1.2.0
